services/galley/test/integration/API/MessageTimer.hs

```diff
@@ -99,6 +100,36 @@ messageTimerChange = do
   getConv jane cid
     !!! const timer1year === (cnvMessageTimer <=< responseJsonUnsafe)
 
+messageTimerChangeQualified :: TestM ()
+messageTimerChangeQualified = do
+  localDomain <- viewFederationDomain
+  -- Create a conversation without a timer
+  [alice, bob, jane] <- randomUsers 3
+  connectUsers alice (list1 bob [jane])
+  rsp <-
+    postConv alice [bob, jane] Nothing [] Nothing Nothing
+      <!! const 201 === statusCode
+  cid <- assertConv rsp RegularConv alice alice [bob, jane] Nothing Nothing
+  let qcid = Qualified cid localDomain
+  -- Set timer to null and observe 204
+  putMessageTimerUpdateQualified alice qcid (ConversationMessageTimerUpdate Nothing)
+    !!! const 204 === statusCode
+  -- Set timer to 1 second
+  putMessageTimerUpdateQualified alice qcid (ConversationMessageTimerUpdate timer1sec)
+    !!! const 200 === statusCode
+  getConv jane cid
+    !!! const timer1sec === (cnvMessageTimer <=< responseJsonUnsafe)
+  -- Set timer to null
+  putMessageTimerUpdateQualified bob qcid (ConversationMessageTimerUpdate Nothing)
+    !!! const 200 === statusCode
+  getConv jane cid
+    !!! const Nothing === (cnvMessageTimer <=< responseJsonUnsafe)
+  -- Set timer to 1 year
+  putMessageTimerUpdateQualified bob qcid (ConversationMessageTimerUpdate timer1year)
+    !!! const 200 === statusCode
+  getConv jane cid
+    !!! const timer1year === (cnvMessageTimer <=< responseJsonUnsafe)
+
```

Comment:
I think this triggers an event, we should capture it and make assertions. Feel free to do it while actually implementing the federation part.

Reply:
Yes, I've done that as part of the followup PR #1783.